common/utils/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -106,6 +106,21 @@ object SparkException {
       messageParameters = Map("message" -> msg),
       cause = cause)
   }
+
+  /**
+   * This is like the Scala require precondition, except it uses SparkException.
```

Comment:
Precisely speaking, `SparkException` and `SparkIllegalArgumentException` are two different exception, even not sub-classes.